services/wallet/saved_addresses.go

```diff
@@ -2,54 +2,195 @@ package wallet
 
 import (
 	"database/sql"
+	"fmt"
+	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 )
 
+type SavedAddressMeta struct {
+	Removed     bool
+	UpdateClock uint64 // wall clock used to deconflict concurrent updates
+}
+
 type SavedAddress struct {
 	Address common.Address `json:"address"`
 	// TODO: Add Emoji and Networks
 	// Emoji    string         `json:"emoji"`
 	Name      string `json:"name"`
 	Favourite bool   `json:"favourite"`
 	ChainID   uint64 `json:"chainId"`
+	SavedAddressMeta
```

Comment:
Do you want to expose this through the API?
If that's the case, you might want to set `json:"savedAddressMeta"`, otherwise I'd make it private (`savedAddressMeta`), if it needs to be public, then you can make sure is not encoded in `JSON` "json:"-"`, I think it needs to be one of the 3 :)